### PR TITLE
Extract Google API accesses from BigQueryOutput and add API error classes

### PR DESCRIPTION
**Pull request by ttanimichi**
It looked like BigQueryOutput was in violation of the Single Responsibility Principle (SRP) because it accesses the Google API a lot, in addition to its responsibilities as a BufferedOutput. So I extracted a class to hold the web access logic.

I also fulfilled a [TODO to add error classes for each API](https://github.com/kaizenplatform/fluent-plugin-bigquery/blob/b76a37ce9c07c3d5dbd34827842ddea742ce94cc/lib/fluent/plugin/out_bigquery.rb#L12-14).
- [x] Add API error classes
- [x] Update README
- [x] Add BigQuery API client class and its test cases
- [x] Fix BigQueryOutput
- [x] Fix existing test cases

### CODE REVIEW

**Comment by ttanimichi** [comment]
I tried hard to maintain backward compatibility, but I couldn't get it 100% right. There are some incompatibilities here and there (e.g. log, exception types, etc.). You may need to bump up the major version to v0.3.0 if this gets merged.

File: test/plugin/test_bigquery_client.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      client.authorization = auth
+    end
+    mock(Google::APIClient).new.with_any_args { client }
+    mock(Google::APIClient::JWTAsserter).new.with_any_args { mock(Object.new).authorize { auth } }
+    mock(Google::APIClient::PKCS12).load_key('/path/to/keyfile.p12', 'itsasecret')
+    expected = [
+      { 'name' => 'time', 'type' => 'TIMESTAMP' },
+      { 'name' => 'tty',  'type' => 'STRING'    }
+    ]
+    assert { @client.fetch_schema('my_table') == expected }
```
Comment (ttanimichi) [inline test/plugin/test_bigquery_client.rb]: We can use Power Assert because we are using test-unit3.

**Comment by ttanimichi** [comment]
I have unWIPed. Please review.

**Comment by ttanimichi** [comment]
I felt there is a problem, but it was groundless :smiley:
I have unWIPed. Please review.

**Comment by naoya** [comment]
Sorry for late respond
Let me give some time.

**Comment by ttanimichi** [comment]
np :)

File: lib/fluent/plugin/bigquery/bigquery_client.rb [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      @client = nil # clear cashed client when errors occur
+      error =
+        case result.status
+        when 404      then NotFound
+        when 409      then Conflict
+        when 400..499 then ClientError
+        when 500..599 then ServerError
+        else UnexpectedError
+        end
+      fail error, result.error_message
```
Comment (ttanimichi) [inline lib/fluent/plugin/bigquery/bigquery_client.rb]: google-api-ruby-client has improved, and we can get error message simply.
ref. https://github.com/google/google-api-ruby-client/commit/1ff1a8e355132fdbb4ab1d7da215fb68216418be

**Comment by ttanimichi** [comment]
How about this PR? :no_mouth:

**Comment by naoya** [comment]
Sorry, will review in this weekend! :bow:

**Pull request closed without merging**